prog_test.go

```diff
@@ -360,6 +360,41 @@ func TestProgramGetNextID(t *testing.T) {
 	}
 }
 
+func TestProgramGetFDByID(t *testing.T) {
+	testutils.SkipOnOldKernel(t, "4.13", "bpf_prog_get_fd_by_id")
+
+	prog, err := NewProgram(&ProgramSpec{
+		Type: SkSKB,
+		Instructions: asm.Instructions{
+			asm.LoadImm(asm.R0, 0, asm.DWord),
+			asm.Return(),
+		},
+		License: "MIT",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer prog.Close()
+	var next ProgramID
+
+	if next, err = ProgramGetNextID(ProgramID(0)); err != nil {
+		t.Fatal("Can't get next ID:", err)
+	}
+	if next == ProgramID(0) {
+		t.Fatal("Expected next ID other than 0")
+	}
+
+	if _, err = NewProgramFromID(next); err != nil {
+		t.Fatalf("Can't get FD for program ID %d: %v", uint32(next), err)
+	}
+
+	// As there can be multiple programs, we use max(uint32) as ProgramID to trigger an expected error.
+	_, err = NewProgramFromID(ProgramID(4294967295))
```

Comment:
Why not `math.MaxUint32`?

Reply:
I didn't want to add another dependency. But you're right, it's easier with `math.MaxUint32`.